mne_bids/read.py

```diff
@@ -700,7 +722,36 @@ def get_matched_empty_room(bids_basename, bids_root):
             space=params.get('space', None)
         )
 
-    return er_basename
+        if 'ses' in params:  # Try to extract date from filename.
+            try:
+                meas_date = datetime.strptime(params['ses'], '%Y%m%d')
+            except (ValueError, TypeError):
+                # There is a session in the filename, but it doesn't encode a
+                # valid date.
+                pass
+
+        if meas_date is None:  # No luck so far! Let's check info['meas_date']
+            _, ext = _parse_ext(path.name)
+            if ext == '.fif':
+                extra_params = dict(allow_maxshield=True)
+            else:
+                extra_params = None
+
+            raw = read_raw_bids(bids_basename=er_basename,
+                                bids_root=bids_root,
+                                extra_params=extra_params)
+
+            meas_date = raw.info['meas_date']
+            if meas_date is None:  # There's nothing we can do.
+                continue
```

Comment:
just an idea, but maybe keep a count of the number of files that you couldn't get meas date for and raise a warning at the end of the loop

Reply:
Good point, will look into this.

Reply:
Implemented this.